caput/memh5.py

```diff
@@ -2226,27 +2470,40 @@ def deep_group_copy(
             if convert_dataset_strings:
                 # Convert unicode strings back into ascii byte strings. This will break
                 # if there are characters outside of the ascii range
-                if isinstance(g2, h5py.Group):
+                if isinstance(g2, file_format.module.Group):
                     data = ensure_bytestring(entry[selection])
 
                 # Convert strings in an HDF5 dataset into unicode
                 else:
                     data = ensure_unicode(entry[selection])
-
-            elif isinstance(g2, h5py.Group):
+            elif isinstance(g2, file_format.module.Group):
                 data = check_unicode(entry)
                 data = data[selection]
             else:
                 data = entry[selection]
 
+            if isinstance(g2, file_format.module.Group):
+                compression_entry = getattr(entry, "compression", None)
+                if compression_entry is not None:
+                    compression = compression_entry
+                    compression_opts = getattr(entry, "compression_opts", None)
+                compression_kwargs = file_format.compression_kwargs(
+                    compression=compression,
+                    compression_opts=compression_opts,
+                    compressor=getattr(entry, "compressor", None),
+                )
+            else:
+                # use HDF5 compression args format
```

Comment:
If this codeblock is going to run for `file_format.Zarr` as well, then we cannot have `HDF5.compression_kwargs` return `{}` by default. I actually do not understand what the `if isinstance(g2, file_format.module.Group)` check is about. Is this, if it is a `dataset`, then we use `HDF5.compression_kwargs`? Does `zarr` have datasets?

Reply:
I removed the return `{}` by default in `HDF5.compression_kwargs` until we understand this better.

Reply:
If `else`, that is the `in-memory` case, and by default Rick uses the HDF5 compression kwarg format for that case.